src/storage/RepresentationConvertingStore.ts

```diff
@@ -69,28 +65,42 @@ export class RepresentationConvertingStore<T extends ResourceStore = ResourceSto
     return this.source.setRepresentation(identifier, representation, conditions);
   }
 
+  /**
+   * Helper function that checks if the given representation matches the given preferences.
+   */
   private matchesPreferences(representation: Representation, preferences: RepresentationPreferences): boolean {
-    if (!preferences.type) {
-      return true;
-    }
     const { contentType } = representation.metadata;
-    return Boolean(
-      contentType &&
-      preferences.type.some((type): boolean =>
-        type.weight > 0 &&
-        matchingMediaType(type.value, contentType)),
-    );
+
+    // Check if there is a result if we try to map the preferences to the content-type
+    return matchingTypes(preferences, [ contentType! ]).length > 0;
```

Comment:
We probably want to avoid using `!` as much as possible, as this disables our TS checks.

Reply:
I admit it might have been me being lazy and not wanting to bother with another test since many things in our code already depend on there being a content-type 😅